albumentations/core/keypoints_utils.py

```diff
@@ -15,16 +18,59 @@
     "filter_keypoints",
     "KeypointsProcessor",
     "KeypointParams",
+    "use_keypoints_ndarray",
 ]
 
 keypoint_formats = {"xy", "yx", "xya", "xys", "xyas", "xysa"}
 
 
-def angle_to_2pi_range(angle: float) -> float:
+def angle_to_2pi_range(angle: Union[np.ndarray, float]):
     two_pi = 2 * math.pi
     return angle % two_pi
 
 
+def split_keypoints_targets(keypoints: Sequence[KeypointType], coord_length: int) -> Tuple[np.ndarray, List[Any]]:
+    kps_array, targets = [], []
+    for kp in keypoints:
+        kps_array.append(kp[:coord_length])
+        targets.append(kp[coord_length:])
+    return np.array(kps_array, dtype=float), targets
+
+
+def use_keypoints_ndarray(return_array: bool = True) -> Callable:
+    """Decorate a function and return a decorator.
+    Since most transformation functions does not alter the amount of bounding boxes, only update the internal
+    keypoints' coordinates, thus this function provides a way to interact directly with
+    the KeypointsInternalType's internal array member.
+
+    Args:
+        return_array (bool): whether the return of the decorated function is a KeypointsArray.
+
+    Returns:
+        Callable: A decorator function.
+    """
+
+    def dec(func: Callable) -> Callable:
+        @wraps(func)
+        def wrapper(
+            keypoints: Union[KeypointsInternalType, np.ndarray], *args, **kwargs
+        ) -> Union[KeypointsInternalType, np.ndarray]:
+            if isinstance(keypoints, KeypointsInternalType):
+                ret = func(keypoints.array, *args, **kwargs)
+                if not return_array:
+                    return ret
+                if not isinstance(ret, np.ndarray):
+                    raise TypeError(f"The return from {func.__name__} must be a numpy ndarray.")
+                keypoints.array = ret
+            elif isinstance(keypoints, np.ndarray):
+                keypoints = func(keypoints.astype(float), *args, **kwargs)
+            return keypoints
```

Comment:
Why are we just returning `keypoints` in the `else` case? I think this behavior can lead to errors.

Reply:
ah yes, I'll raise TypeErrors in the else statement then. Thanks for pointing this out.